File: src/content.service.ts
import { Content } from './content.interface';
import { DataService } from './data.service';
import { HttpClient } from './http.service';

export class ContentService extends DataService<Content> {
  constructor(httpClient: HttpClient) {
    super(httpClient, 'api/contents');
  }

  download = (id: string, raw: boolean = false): Promise<Blob | ArrayBuffer> => {
    if (raw) {
      return this.httpClient.get<Blob>(`${this.basePath}/${id}/download`, { responseType: 'arraybuffer' });
    } else {
      return this.httpClient.get<Blob>(`${this.basePath}/${id}/download`, { responseType: 'blob' });
    }
  };
}

File: packages/hahnpro/hpc_api/hpc_api/content_service.py
from .http_client import HttpClient


class ContentService:

    def __init__(self, http_client: HttpClient):
        self.http_client = http_client
        self.base_path = '/contents'

    def get_one(self, id: str, options={}):
        params = {} if ('populate' not in options) else {"populate": options["populate"]}
        return self.http_client.get(f'{self.base_path}/{id}', params).json()

    def delete_one(self, id):
        return self.http_client.delete(f'{self.base_path}/{id}').json()

    def download(self, id: str):
        return self.http_client.get(f'{self.base_path}/{id}/download', {}).content
